hbp_validation_framework/__init__.py

```diff
@@ -308,53 +996,734 @@ def get_platform(self):
+        """Edit an existing model instance.
+
+        This allows to edit an instance of an existing model in the model catalog.
+        The model instance can be specified in the following ways (in order of priority):
+
+        1. specify `instance_id` corresponding to model instance in model catalog
+        2. specify `model_id` and `version`
+        3. specify `alias` (of the model) and `version`
+
+        Parameters
+        ----------
+        instance_id : UUID
+            System generated unique identifier associated with model instance.
+        model_id : UUID
+            System generated unique identifier associated with model description.
+        alias : string
+            User-assigned unique identifier associated with model description.
+        source : string
+            Path to model source code repository (e.g. github).
+        version : string
+            User-assigned identifier (unique for each model) associated with model instance.
+        parameters : string
+            Any additional parameters to be submitted to model at runtime.
+
+        Returns
+        -------
+        UUID
+            UUID of the model instance that has been created.
+
+        Examples
+        --------
+        >>> instance_id = model_catalog.edit_model_instance(instance_id="fd1ab546-80f7-4912-9434-3c62af87bc77",
+                                                model_id="196b89a3-e672-4b96-8739-748ba3850254",
+                                                source="https://www.abcde.com",
+                                                version="10.0",
+                                                parameters="")
+        """
+
+        id = instance_id
+        instance_data = locals()
+        for key in ["self", "instance_id"]:
+            instance_data.pop(key)
+
+        if instance_id == "" and (model_id == "" or version == "") and (alias == "" or version == ""):
+            raise Exception("instance_id or (model_id, version) or (alias, version) needs to be provided for finding a model instance.")
+        elif instance_id:
+            url = self.url + "/scientificmodelinstance/?id=" + instance_id + "&format=json"
+            #url = self.url + "/scientificmodelinstance/?format=json"
+        elif model_id and version:
+            url = self.url + "/scientificmodelinstance/?model_id=" + model_id + "&version=" + version + "&format=json"
+        else:
+            url = self.url + "/scientificmodelinstance/?test_alias=" + alias + "&version=" + version + "&format=json"
+
+        headers = {'Content-type': 'application/json'}
+        response = requests.put(url, data=json.dumps([instance_data]), auth=self.auth, headers=headers)
+        if str(response) == "<Response [202]>":
+            return response.json()
+        else:
+            raise Exception("Error in editing model instance. Response = " + str(response.json()))
+
+    def get_model_image(self, image_id=""):
```

Comment:
"image" is slightly ambiguous in this context; it could conceivably be taken to mean a virtual machine or Docker image.

Reply:
Updated docs to indicate images are simply figures (as discussed on slack).